src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java

```diff
@@ -1847,6 +1851,57 @@ void testAssertItemConsumedWithNullItem()
 		assertThrows(NullPointerException.class, () -> player.assertItemConsumed(null));
 	}
 
+	@Test
+	void testSimulateConsumeItemStatusEffectIsAppliedFromItem()
+	{
+		ItemStack edibleWithStatusEffect = new ItemStack(Material.GOLDEN_APPLE);
+		player.simulateConsumeItem(edibleWithStatusEffect);
+		assertNotNull(player.getPotionEffect(PotionEffectType.REGENERATION));
+	}
+
+	@Test
+	void testSimulateConsumeItemStatusEffectWithProbabilityIsAlwaysApplied()
+	{
+		ItemStack edibleWithStatusEffectWithProbability = new ItemStack(Material.CHICKEN);
+		player.simulateConsumeItem(edibleWithStatusEffectWithProbability, true);
+		assertNotNull(player.getPotionEffect(PotionEffectType.HUNGER));
+	}
+
+	@Test
+	void testSimulateConsumeItemStatusEffectWithProbabilityIsNotApplied()
+	{
+		ItemStack edibleWithStatusEffectWithProbability = new ItemStack(Material.CHICKEN);
+		player.simulateConsumeItem(edibleWithStatusEffectWithProbability, false);
+		assertNull(player.getPotionEffect(PotionEffectType.HUNGER));
+	}
+
+	@ParameterizedTest
+	@MethodSource("potionItemProvider")
+	void testSimulateConsumePotionItemWithBaseEffectIsApplied(ItemStack potion, PotionEffect inflictedEffect) {
+		player.simulateConsumeItem(potion);
+		assertEquals(inflictedEffect, player.getPotionEffect(inflictedEffect.getType()));
+	}
+
+	private static Stream<Arguments> potionItemProvider() {
+		MockBukkit.getOrCreateMock();
+		return Stream.of(
+				Arguments.of(
+						potionItemStack(PotionType.REGENERATION), new PotionEffect(PotionEffectType.REGENERATION, 900, 0, false, true, true),
+						potionItemStack(PotionType.LONG_REGENERATION), new PotionEffect(PotionEffectType.REGENERATION, 1800, 0, false, true, true),
+						potionItemStack(PotionType.STRONG_REGENERATION), new PotionEffect(PotionEffectType.REGENERATION, 450, 1, false, true, true),
+						potionItemStack(PotionType.AWKWARD), null
+				)
+		);
+	}
+
+	private static ItemStack potionItemStack(PotionType potionType) {
+		ItemStack itemStack = new ItemStack(Material.POTION);
+		PotionMeta potionMeta = (PotionMeta) itemStack.getItemMeta();
+		potionMeta.setBasePotionType(potionType);
+		itemStack.setItemMeta(potionMeta);
+		return itemStack;
+	}
```

Comment:
This fails because `getItemMeta` requires the mocked server. Usually adding `MockBukkit#getOrCreate()` is enough, but because this test class uses its own `ServerMock` implementation, I don't know how to handle this, since I cannot just change the call there to `MockBukkit#getOrCreate()`.

Reply:
I see, there's two simple approaches here (there's probably many more):
1. Only test one food item in a non parameterized test
2. Make each argument in the methodsource be a [supplier](https://docs.oracle.com/javase/8/docs/api/java/util/function/Supplier.html), which you can use to supply all the necessary data

I think 1 is good enough here, as 2 feels like it becomes a bit messy